lib/foods.js

```diff
@@ -1,65 +1,75 @@
+    $("tbody").append(localStorage.getItem(localStorage.key(i)));
+    add_delete_functionality();
+  }
+
   $("#add-food").on('click', function() {
     clear_errors();
 
     var name = $('#name-field input').val();
     var calories = $('#calories-field input').val();
-    
-    
+
+
     check_for_name(name)
     check_for_calories(calories)
     if (name == "" || calories == "") {
       return 1;
     }
 
     var foodRow = `<tr class='food-row'>
-                      <td class='food-name'>` + name + `</td>
-                      <td class='food-calories'>` + calories + `</td>
-                      <td class='food-delete'><button>-</button></td>
-                    </tr>`
+    <td class='food-name'>` + name + `</td>
+    <td class='food-calories'>` + calories + `</td>
+    <td class='food-delete'><button>-</button></td>
+    </tr>`
+    localStorage.setItem(name, foodRow);
     $("tbody").prepend(foodRow);
     add_delete_functionality();
     clear_text();
-  })
-  $("#food-filter input").on('keyup', function(){
-    var filter = $("#food-filter input").val().toLowerCase();
-    var foods = $("tbody").children();
-    for (var i = 0; i< foods.length; i++) {
-      var food = $(foods[i]).find('.food-name').text().toLowerCase();
-      if (!food.contains(filter)) {
-        $(foods[i]).slideUp();
-      }
-      else  {
-        $(foods[i]).slideDown();
+
+
+
+
+    $("#food-filter input").on('keyup', function(){
+      var filter = $("#food-filter input").val().toLowerCase();
+      var foods = $("tbody").children();
+      for (var i = 0; i< foods.length; i++) {
+        var food = $(foods[i]).find('.food-name').text().toLowerCase();
+        if (!food.contains(filter)) {
+          $(foods[i]).slideUp();
+        }
+        else  {
+          $(foods[i]).slideDown();
+        }
       }
-    }
+    })
   })
-})
 
-function check_for_name(name) {
-  if (name == ""){
+  function check_for_name(name) {
+    if (name == ""){
       $('#name-field .validation-error').html('Please Enter a Name');
     }
-}
+  }
 
-function check_for_calories(calories) {
-  if (calories == ""){
+  function check_for_calories(calories) {
+    if (calories == ""){
       $('#calories-field .validation-error').html('Please Enter Calories');
     }
-}
+  }
 
-function clear_errors(){
-  $('#name-field .validation-error').empty();
-  $('#calories-field .validation-error').empty();
-}
+  function clear_errors(){
+    $('#name-field .validation-error').empty();
+    $('#calories-field .validation-error').empty();
+  }
 
-function clear_text(){
-  $('#name-field input').val(null);
-  $('#calories-field input').val(null);
-}
+  function clear_text(){
+    $('#name-field input').val(null);
+    $('#calories-field input').val(null);
+  }
 
-function add_delete_functionality(){
-  $("td.food-delete button").on('click', function(){
+  function add_delete_functionality(){
+    $("td.food-delete button").on('click', function(){
       $(this).parent().parent().remove();
```

Comment:
@case-eee I had mentioned doing something like
```
$("#food-list").on("click", "button", function() {
    $(this).closest("tr").remove();
```
is this more like what you were thinking?